### PR TITLE
Benchmark tests

### PR DESCRIPTION
**Pull request by aojea**
## Why is this PR needed?

It allows running benchmark tests, that may be useful if we need to test our code at scale.

## What does this PR do?

Adds a new option `make test-bench` to the Makefile so it can runs benchmark tests, and runs it automatically when using `make test`
It also implements an example benchmarking the function `getPodFromList()`

## Anything else a reviewer needs to know?

Special test cases, manual steps, links to resources or anything else that could be helpful to the reviewer.

## Info for QA

This is info for QA so that they can validate this. This is **mandatory** if this PR fixes a bug.
If this is a new feature, a good description in "What does this PR do" may be enough.

### Related info

Info that can be relevant for QA:
* link to other PRs that should be merged together
* link to packages that should be released together
* upstream issues

### Status **BEFORE** applying the patch

How can we reproduce the issue? How can we see this issue? Please provide the steps and the prove
this issue is not fixed.

### Status **AFTER** applying the patch

How can we validate this issue is fixed? Please provide the steps and the prove this issue is fixed.

## Docs

If docs need to be updated, please add a link to a PR to https://github.com/SUSE/doc-caasp.
At the time of creating the issue, this PR can be work in progress (set its title to [WIP]),
but the documentation needs to be finalized before the PR can be merged. 

# Merge restrictions

(Please do not edit this)

We are in *v4-maintenance phase*, so we will restrict what can be merged to prevent unexpected surprises:

    What can be merged (merge criteria):
        2 approvals:
            1 developer: code is fine
            1 QA: QA is fine
        there is a PR for updating documentation (or a statement that this is not needed)

### CODE REVIEW

**Comment by Itxaka** [comment]
Looks great!

Should we track the benchmarking in jenkins? That way we can see how it evolves over time and how changes affect the performance of the code :)

File: internal/pkg/skuba/kubernetes/pod_test.go [hunk 1]
```diff
@@ -18,10 +18,12 @@
 package kubernetes
 
 import (
+	"fmt"
 	"reflect"
 	"testing"
 
 	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (c3y1huang) [inline internal/pkg/skuba/kubernetes/pod_test.go]: Duplicates?

Comment (aojea) [reply internal/pkg/skuba/kubernetes/pod_test.go]: good catch

VScode autocomplete :man_shrugging:

Comment (aojea) [reply internal/pkg/skuba/kubernetes/pod_test.go]: solved

**Pull request merged**